volatility3/framework/plugins/linux/pidhashtable.py

```diff
@@ -0,0 +1,249 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import List
+
+from volatility3.framework import renderers, interfaces, constants
+from volatility3.framework.symbols import linux
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.configuration import requirements
+from volatility3.plugins.linux import pslist
+
+vollog = logging.getLogger(__name__)
+
+
+class PIDHashTable(plugins.PluginInterface):
+    """Enumerates processes through the PID hash table"""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="linuxutils", component=linux.LinuxUtilities, version=(2, 2, 0)
+            ),
+            requirements.BooleanRequirement(
+                name="decorate_comm",
+                description="Show `user threads` comm in curly brackets, and `kernel threads` comm in square brackets",
+                optional=True,
+                default=False,
+            ),
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.vmlinux = None
+        self.vmlinux_layer = None
+
+    def _is_valid_task(self, task):
+        return task and task.pid > 0 and self.vmlinux_layer.is_valid(task.parent)
+
+    def _get_pidtype_pid(self):
+        # The pid_type enumeration is present since 2.5.37, just in case
+        pid_type_enum = self.vmlinux.get_enumeration("pid_type")
```

Comment:
I'd much prefer this parameterized than maintaining state.  I can't fortell if someone's going to reuse plugins in a different UI somehow, so maintaining it by passing it through parameters rather than state seems better and would allow these functions to be exposed later on if they prove useful for others.